tests/unit-tests/signal_tests/deterministic/alarm.c

```diff
@@ -0,0 +1,34 @@
+// test for alarm function
```

Comment:
We should probably have a signal test for each signal type. A bit tedious but will be useful once the CI kicks in.

Reply:
We can create a separate issue for this if you want though, don't need to handle it here explicitly.